scripts/sync-readme-to-ecr-public.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+repo_root_path="$(cd "$(dirname "$0")"; cd ..; pwd -P)"
+
+if ! command -v jq; then
+    echo "command not found: jq" >&2
+    exit 1
+fi
+
+region="${AWS_REGION:-us-east-1}"
+repo_root="${KO_DOCKER_REPO}"
+version="$(make -s -f "${makefile_path}" version)"
+
+usage=$(cat << EOM
+usage: $(basename $0) -h | [-g REGION] [-r REPOSITORY] [-v VERSION]
+
+    Upload the README.md to the AWS ECR controller and webhook repositories.
+
+    Options:
+        -h             Display this help message then exit.
+        -g REGION      AWS Region of ECR image repository. Defaults to "${region}".
+        -r REPOSITORY  Image repository to push the built images. Defaults to "${repo_root}".
+        -v VERSION     Node Termination Handler version. Defaults to "${version}".
+
+EOM
+)
+
+while getopts "r:h" opt; do
+    case "${opt}" in
+        g ) region="${OPTARG}"
+            ;;
+        r ) repo_root="${OPTARG}"
+            ;;
+        v ) version="${OPTARG}"
+            ;;
+        h ) echo "${usage}"
+            exit 0
+            ;;
+        \?) echo "${usage}" >&2
+            exit 1
+            ;;
+    esac
+done
+
+assert_not_empty() {
+    if [[ -z "${!1}" ]]; then
+        echo "error: missing argument ${1}" >&2
+        echo "${usage}" >&2
+        exit 1
+    fi
+}
+
+assert_not_empty region
+assert_not_empty repo_root
+assert_not_empty version
+
+#################################################
+
+if ! (git --no-pager diff --name-only HEAD^ HEAD | grep 'README.md' >/dev/null); then
```

Comment:
The latent assumption here seems to be that the README, if it was modified at all since the prior release, will have been modified _at least_ in the most recent commit. In our current release process, that is generally true, but it is not obvious unless we read the script itself (perhaps a bit of a leaky abstraction).

This is probably okay, since a failure in which the README was modified (but not in the latest commit) will likely be fixed during the next release, even if we don't notice the failure at the time. Just wanted to clarify and see if you agree about that it is safe to make this assumption, rather than engineering a more robust solution.

Reply:
I copied this logic from the v1 script, but I agree with your point.  I'll update it to use the latest and previous release tags.